src/DurableTask.Netherite/StorageLayer/Faster/MemoryTracker.cs

```diff
@@ -41,7 +41,8 @@ public CacheTracker NewCacheTracker(FasterKV store, int partitionId, CacheDebugg
 
         public void UpdateTargetSizes()
         {
-            if (this.stores.Count > 0)
+            int numberOfStores = this.stores.Count;
+            if (numberOfStores > 0)
```

Comment:
nit: let's document the race condition, since it's non-obious
```suggestion
            // store count to avoid race condition where count goes to zero and causes divide-by-zero error
            int numberOfStores = this.stores.Count;
            if (numberOfStores > 0)
```